Code/Framework/AzFramework/AzFramework/Physics/Utils.h

```diff
@@ -36,9 +36,15 @@ namespace Physics
         using NameSet = AZStd::unordered_set<AZStd::string>;
 
         /// Helper routine for certain physics engines that don't directly expose this property on rigid bodies.
-        AZ_INLINE AZ::Matrix3x3 InverseInertiaLocalToWorld(const AZ::Vector3& diag, const AZ::Matrix3x3& rotationToWorld)
+        AZ_INLINE AZ::Matrix3x3 InverseInertiaLocalToWorld(const AZ::Vector3& inverseInertiaDiagonal, const AZ::Matrix3x3& rotationToWorld)
         {
-            return rotationToWorld * AZ::Matrix3x3::CreateDiagonal(diag) * rotationToWorld.GetTranspose();
+            return rotationToWorld * AZ::Matrix3x3::CreateDiagonal(inverseInertiaDiagonal) * rotationToWorld.GetTranspose();
+        }
+
+        /// Helper routine for certain physics engines that don't directly expose this property on rigid bodies.
+        AZ_INLINE AZ::Matrix3x3 InertiaLocalToWorld(const AZ::Vector3& inertiaDiagonal, const AZ::Matrix3x3& rotationToWorld)
+        {
+            return rotationToWorld * AZ::Matrix3x3::CreateDiagonal(inertiaDiagonal) * rotationToWorld.GetTranspose();
```

Comment:
These 2 functions do exactly the same operations to its input parameters. Maybe remove `InverseInertiaLocalToWorld` and leave only `InertiaLocalToWorld`?

Reply:
I deliberately didn't do that to avoid confusion. Even though these two functions do the same thing, this bug happened because inertia and inverse inertia were getting mixed up. I guess I could replace both with a DiagonalMatrixLocalToWorld function or something more generically named.

Reply:
Finding the right name for it and avoid code duplication seems good to me. Thanks.